### PR TITLE
[HNC-573] - Modify Release Workflow UI for pdi-plugins-ee

### PR DESCRIPTION
**Pull request by mpadhi1**
HNC-573 Changes for publishing release bundles.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -176,46 +176,46 @@ jobs:
...
+          --release_method "artifacts"
       
-      - name: Push artifacts to Box
-        if: ${{ inputs.box_upload == true }}
-        shell: bash
-        run: |
-          python3 -m venv env
-          source env/bin/activate
-          pip3 install -q pyyaml dohq-artifactory requests boxsdk tqdm
+      # - name: Push artifacts to Box
```
Comment (cardosov) [inline .github/workflows/release.yml:185]: We are not to do Box upload anymore?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -153,7 +153,7 @@ jobs:
         with:
           repository: pentaho/actions-common
           path: actions-common
-          ref: stable
+          ref: HNC-573
```
Comment (cardosov) [inline .github/workflows/release.yml:156]: This should continue to be `stable`, right?

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -132,7 +132,7 @@ jobs:
             --user ${{ env.NEXUS_DEPLOY_USER }}
 
       - name: Promote artifacts
-        if: ${{ inputs.promote_artifacts == true }}
+        # if: ${{ inputs.promote_artifacts == true }}
```
Comment (cardosov) [inline .github/workflows/release.yml:135]: Wasn't the promote artifacts step suppose to switch position with the upload do Box?

**Review by cardosov — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 2 above)
Comment (mpadhi1) [reply .github/workflows/release.yml:156]: yes it should be stable so that we merge it.

File: .github/workflows/release.yml (hunk 1 above)
Comment (mpadhi1) [reply .github/workflows/release.yml:185]: the ticket said to remove the box upload parameters.

Comment (cardosov) [reply .github/workflows/release.yml:185]: Ok but this is all the code to do the upload itself!

File: .github/workflows/release.yml (hunk 3 above)
Comment (mpadhi1) [reply .github/workflows/release.yml:135]: I am not sure if we need to promote artefacts at release bundle step

Comment (mpadhi1) [reply .github/workflows/release.yml:135]: We are not promoting artefacts while release bundle creation.

**Review by mayur-hitachivantara — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 3 above)
Comment (cardosov) [reply .github/workflows/release.yml:135]: Why?

File: .github/workflows/release.yml (hunk 1 above)
Comment (mpadhi1) [reply .github/workflows/release.yml]: changes done!!

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -153,7 +153,7 @@ jobs:
         with:
           repository: pentaho/actions-common
           path: actions-common
-          ref: stable
+          ref: HNC-573
 
       - name: Create Release Bundle
         if: ${{ inputs.create_release_bundle == true }}
```
[the remark this answers is not in the document]
Comment (mpadhi1) [reply .github/workflows/release.yml:159]: as box upload and promote artefacts are now enabled, keeping this if option available.

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -32,36 +32,36 @@ on:
         type: boolean
         default: true
         description: "Dry run. No real changes should occur."
-      promote_artifacts:
-        type: boolean
-        default: true
-        description: "Promote artifacts in Artifactory."
+      # promote_artifacts:
```
Comment (cardosov) [inline .github/workflows/release.yml:35]: This parameter is in use. Can't be commented out

**Review by cardosov — CHANGES_REQUESTED** [review]
Update the title of the PR with the convention of `[TICKET-NBR] - Ticket description`. Also, squash the commits

**Review by mayur-hitachivantara — APPROVED** [review]

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -41,27 +41,27 @@ on:
...
-      signing_key_name:
-        description: "The GPG/RSA key-pair name given in Artifactory."
-        required: false
-        type: string 
-      release_method:
-        description: "Source type to create release bundle (i.e builds or artifacts)."
-        required: false
-        default: artifacts
-        type: string       
+      # release_bundle_name:
```
Comment (cardosov) [inline .github/workflows/release.yml:44]: If these params are not needed, don't leave commented code in here.

File: .github/workflows/release.yml [hunk 7]
```diff
@@ -41,27 +41,27 @@ on:
...
+      # release_method:
+      #   description: "Source type to create release bundle (i.e builds or artifacts)."
+      #   required: false
+      #   default: artifacts
+      #   type: string       
       box_upload:
-        type: boolean
-        default: true
-        description: "Box upload."
+         type: boolean
```
Comment (cardosov) [inline .github/workflows/release.yml:62]: Unneeded extra spaces

File: .github/workflows/release.yml [hunk 8]
```diff
@@ -190,7 +148,7 @@ jobs:
           source env/bin/activate
           pip3 install -q pyyaml dohq-artifactory requests boxsdk tqdm
           
-          echo ${{ env.BUILD_NAME }} ${{ env.BUILD_NUMBER }} ${{ env.BUILD_VERSION }}
+           echo ${{ env.BUILD_NAME }} ${{ env.BUILD_NUMBER }} ${{ env.BUILD_VERSION }}
```
Comment (cardosov) [inline .github/workflows/release.yml:151]: extra unnneeded white space

File: .github/workflows/release.yml [hunk 9]
```diff
@@ -216,6 +174,49 @@ jobs:
           if [ "${{ inputs.dry_run }}" = "true" ]; then  
             box_summary="[Dry run] "
           fi
+      - name: Promote artifacts
```
Comment (cardosov) [inline .github/workflows/release.yml:177]: Give an extra line to separate these 2 steps

File: .github/workflows/release.yml [hunk 10]
```diff
@@ -216,6 +174,49 @@ jobs:
...
+          --dry_run ${{ inputs.dry_run }} \
+          --logging_level ${{inputs.logging_level }} \
+          --release_bundle_name ${{ env.BUILD_NAME }} \
+          --release_bundle_version ${{ env.BUILD_NUMBER }} \
+          --signing_key_name "frog" \
+          --arti_host ${{ env.ARTIFACTORY_HOST }} \
+          --release_method "artifacts"
           # BOX_FOLDER_ID is an env var that is set inside the promote-release-to-box.py script file
-          echo "$box_summary :ballot_box: ${{ inputs.release_version }} was uploaded to https://pentaho.app.box.com/folder/$BOX_FOLDER_ID" >> $GITHUB_STEP_SUMMARY
+          # echo "$box_summary :ballot_box: ${{ inputs.release_version }} was uploaded to https://pentaho.app.box.com/folder/$BOX_FOLDER_ID" >> $GITHUB_STEP_SUMMARY
```
Comment (cardosov) [inline .github/workflows/release.yml:222]: These lines belong to the upload to box step and should be there

**Review by cardosov — CHANGES_REQUESTED** [review]
Review my comment about the PR title and the commits squashing

File: .github/workflows/release.yml (hunk 6 above)
Comment (mpadhi1) [reply .github/workflows/release.yml:44]: removed commented code

File: .github/workflows/release.yml (hunk 8 above)
Comment (mpadhi1) [reply .github/workflows/release.yml:151]: removed extra space

File: .github/workflows/release.yml (hunk 9 above)
Comment (mpadhi1) [reply .github/workflows/release.yml:163]: added extra line

**Review by mpadhi1 — COMMENTED** [review]
made necessary changes

File: .github/workflows/release.yml (hunk 7 above)
Comment (cardosov) [reply .github/workflows/release.yml:45]: The extra spaces is still there.

File: .github/workflows/release.yml [hunk 11]
```diff
@@ -216,6 +155,50 @@ jobs:
           if [ "${{ inputs.dry_run }}" = "true" ]; then  
             box_summary="[Dry run] "
           fi
+       # BOX_FOLDER_ID is an env var that is set inside the promote-release-to-box.py script file
+          # echo "$box_summary :ballot_box: ${{ inputs.release_version }} was uploaded to https://pentaho.app.box.com/folder/$BOX_FOLDER_ID" >> $GITHUB_STEP_SUMMARY
```
Comment (cardosov) [inline .github/workflows/release.yml:159]: This echo must be uncommented. It's relevant information to show in the summary

**Review by cardosov — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 11 above)
Comment (mpadhi1) [reply .github/workflows/release.yml:159]: uncommented

File: .github/workflows/release.yml (hunk 10 above)
Comment (mpadhi1) [reply .github/workflows/release.yml]: done

File: .github/workflows/release.yml [hunk 12]
```diff
@@ -131,56 +113,13 @@ jobs:
...
-          ref: stable
-
-      - name: Create Release Bundle
-        if: ${{ inputs.create_release_bundle == true }}
-        shell: bash
-        run: |
-          python3 -m venv env
-          source env/bin/activate
-          pip3 install -q pyyaml dohq-artifactory requests
+          ref: HNC-573
```
Comment (cardosov) [inline .github/workflows/release.yml:121]: Should be `stable`

File: .github/workflows/release.yml (hunk 8 above)
Comment (cardosov) [reply .github/workflows/release.yml:132]: I still see the extra space just before the `echo`

File: .github/workflows/release.yml [hunk 13]
```diff
@@ -216,6 +155,49 @@ jobs:
           if [ "${{ inputs.dry_run }}" = "true" ]; then  
             box_summary="[Dry run] "
           fi
+              echo "$box_summary :ballot_box: ${{ inputs.release_version }} was uploaded to https://pentaho.app.box.com/folder/$BOX_FOLDER_ID" >> $GITHUB_STEP_SUMMARY
```
Comment (cardosov) [inline .github/workflows/release.yml:158]: Can't we align the `echo` with the `if`?

**Review by cardosov — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml [hunk 14]
```diff
@@ -216,6 +155,49 @@ jobs:
           if [ "${{ inputs.dry_run }}" = "true" ]; then  
             box_summary="[Dry run] "
           fi
+            echo "$box_summary :ballot_box: ${{ inputs.release_version }} was uploaded to https://pentaho.app.box.com/folder/$BOX_FOLDER_ID" >> $GITHUB_STEP_SUMMARY
```
Comment (cardosov) [inline .github/workflows/release.yml:158]: Please align this `echo` with the above  `if`.

File: .github/workflows/release.yml [hunk 15]
```diff
@@ -216,6 +155,49 @@ jobs:
...
+          --rt_base_url ${{ env.ARTIFACTORY_BASE_URL }} \
+          --jf_cli_rt_name ${{ inputs.jf_cli_rt_name }} \
+          --dry_run ${{ inputs.dry_run }} \
+          --logging_level ${{inputs.logging_level }} \
+          --release_bundle_name ${{ env.BUILD_NAME }} \
+          --release_bundle_version ${{ env.BUILD_NUMBER }} \
+          --signing_key_name "frog" \
+          --arti_host ${{ env.ARTIFACTORY_HOST }} \
+          --release_method "artifacts"
+
```
Comment (cardosov) [inline .github/workflows/release.yml:203]: The last line shouldn't have any spaces. Please check

**Review by cardosov — CHANGES_REQUESTED** [review]
You have 14 commits. Please squash them into 1 only

File: .github/workflows/release.yml (hunk 13 above)
Comment (mpadhi1) [reply .github/workflows/release.yml]: aligned

File: .github/workflows/release.yml (hunk 14 above)
Comment (mpadhi1) [reply .github/workflows/release.yml:158]: changed

File: .github/workflows/release.yml (hunk 15 above)
Comment (mpadhi1) [reply .github/workflows/release.yml:203]: done

File: .github/workflows/release.yml (hunk 14 above)
Comment (cardosov) [reply .github/workflows/release.yml:158]: I still see it not aligned (posting image):
image

File: .github/workflows/release.yml [hunk 16]
```diff
@@ -216,6 +155,49 @@ jobs:
...
```
Comment (cardosov) [inline .github/workflows/release.yml:203]: Still seeing the some white space in the last line (posting image)
image

**Review by cardosov — CHANGES_REQUESTED** [review]

**Pull request closed without merging**